src/sentry/integrations/slack/message_builder/notifications.py

```diff
@@ -1,69 +1,119 @@
-import re
-from typing import Any, Mapping
+from typing import Any, Mapping, Union
 from urllib.parse import urljoin
 
 from sentry.integrations.slack.message_builder import SlackBody
 from sentry.integrations.slack.message_builder.base.base import SlackMessageBuilder
-from sentry.integrations.slack.message_builder.issues import SlackIssuesMessageBuilder
+from sentry.integrations.slack.message_builder.issues import (
+    SlackIssuesMessageBuilder,
+    build_attachment_title,
+    build_rule_url,
+    get_title_link,
+)
+from sentry.models import Team, User
+from sentry.notifications.activity.assigned import AssignedActivityNotification
+from sentry.notifications.activity.note import NoteActivityNotification
+from sentry.notifications.activity.regression import RegressionActivityNotification
+from sentry.notifications.activity.release import ReleaseActivityNotification
+from sentry.notifications.activity.resolved import ResolvedActivityNotification
+from sentry.notifications.activity.resolved_in_release import ResolvedInReleaseActivityNotification
+from sentry.notifications.activity.unassigned import UnassignedActivityNotification
 from sentry.notifications.base import BaseNotification
 from sentry.notifications.rules import AlertRuleNotification
 from sentry.utils.http import absolute_uri
 
+from ..utils import build_notification_footer, get_referrer_qstring
 
-def get_referrer_qstring(notification: BaseNotification) -> str:
-    return "?referrer=" + re.sub("Notification$", "Slack", notification.__class__.__name__)
-
-
-def get_settings_url(notification: BaseNotification) -> str:
-    return str(
-        urljoin(
-            absolute_uri("/settings/account/notifications/"), get_referrer_qstring(notification)
-        )
-    )
+ISSUE_UNFURL = (
```

Comment:
I think you should replace this with a property like `is_message_issue_unfurl` on `BaseNotification` that says if we should treat the Slack message as an issue unfurl.